qiskit/utils/quantum_instance.py

```diff
@@ -619,13 +619,16 @@ def _find_save_state(data):
                 else:
                     circuits[0:0] = cal_circuits
                     prepended_calibration_circuits = len(cal_circuits)
+                    cal_run_config = copy.deepcopy(self.run_config)
+                    if hasattr(cal_run_config, "parameterizations"):
```

Comment:
I would flip these lines, so no deepcopy if unnecessary.

Reply:
I am not sure, but I feel like since the only time we need to rewrite run_config is when caliburation runs, we need a deepcopy for other cases. Any better code would be welcome.

Reply:
Ah! I understood the "flip" now. That's great!

Reply:
yeap.. sorry. Probably swap would had been a better word.